src/Parallel/InitializationFunctions.cpp

```diff
@@ -14,15 +17,21 @@ void setup_error_handling() {
       try {
         std::rethrow_exception(exception);
       } catch (std::exception& ex) {
-        ERROR("Terminated due to an uncaught exception: " << ex.what());
+        std::ostringstream os;
+        os << "Terminated due to an uncaught exception:\n " << ex.what();
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
+        CkError("%s\n", os.str().c_str());
       } catch (...) {
-        ERROR("Terminated due to unknown exception\n");
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
+        CkError("Terminated due to unknown exception\n");
       }
     } else {
-      ERROR(
+      // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
+      CkError(
           "Terminate was called for an unknown reason (not an uncaught "
           "exception), calling Charm++'s abort function to properly "
           "terminate execution.");
     }
+    sys::abort("");
```

Comment:
This is Charm++'s default implementation:

```cpp
  std::set_terminate([](){
    std::exception_ptr exptr = std::current_exception();
    if (exptr)
    {
      try
      {
        std::rethrow_exception(exptr);
      }
      catch (std::exception &ex)
      {
        CkAbort("Unhandled C++ exception in user code: %s.\n", ex.what());
      }
    }
    else
    {
      CkAbort("Unhandled C++ exception in user code.\n");
    }
  });
```

https://github.com/UIUC-PPL/charm/blob/a167573fff954c49f4180f161fde44cd1de26914/src/ck-core/init.C#L1306

Would it be sufficient to just use that, meaning we can delete `setup_error_handling` altogether?

Reply:
I don't know.   I know our initialization functions are executed on every core, and I'd rather not rely on charm++ in order to see an error message

Reply:
We ultimately added our own `std::terminate` because the Charm++ one wasn't always reporting the exceptions. It might have been updated since we wrote that code, but I agree with Larry. The current thing works and we have good control, so I'd rather keep it as-is